src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            this.elementData = (T[]) new Object[initialCapacity];
```

Comment:
so, count you have array with size 0? How will it be increased? 
```suggestion
        if (initialCapacity >= DEFAULT_CAPACITY) {
            this.elementData = (T[]) new Object[initialCapacity];
         } else if (initialCapacity > 0) {
            this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
```

Reply:
I really didn't keep in mind that. Thanks!